src/realm/list.hpp

```diff
@@ -144,8 +144,9 @@ class Lst final : public CollectionBaseImpl<LstBase, Lst<T>> {
     template <typename Func>
     void find_all(value_type value, Func&& func) const
     {
-        if (update())
+        if (update()) {
             m_tree->find_all(value, std::forward<Func>(func));
```

Comment:
Lst<Mixed>.find_all(Mixed{}) is still not working, it should find every null or unresolved link.

Reply:
As per my previous comments, it does work, the reason is that we are doing a comparison based on type rather than based on actual content for Mixed links